src/main/java/com/alex/eshop/eshop/dto/ItemDTO.java

```diff
@@ -0,0 +1,51 @@
+package com.alex.eshop.eshop.dto;
+
+import com.alex.eshop.eshop.entity.Category;
+
+public class ItemDTO {
+    private Long id;
+    private String name;
+    private Category category;
```

Comment:
`private Category category;`
You must not use Entity object in DTO.
Use `CategoryDTO` or single fields `String categoryName; Long categoryId;`